From 768f2a3213b56295280ad3ef8383fdeac2ecab75 Mon Sep 17 00:00:00 2001
From: Vignesh Viswanathan <quic_viswanat@quicinc.com>
Date: Tue, 1 Aug 2023 12:07:39 +0530
Subject: [PATCH] remoteproc: qcom: q6v5: Get WCSS Crash reason from remote
 SMEM region

WCSS crash reason is allocated by Q6 in the SMEM remote region with id
1. Get the crash reason from remote_id region instead of SMEM global
region.

Change-Id: Ie52b23516737d5eb44e0c571fa3cae00c4b7c1eb
Signed-off-by: Vignesh Viswanathan <quic_viswanat@quicinc.com>
---
 drivers/remoteproc/qcom_q6v5.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/remoteproc/qcom_q6v5.c b/drivers/remoteproc/qcom_q6v5.c
index 5bf6028bbc28..d121a089f933 100644
--- a/drivers/remoteproc/qcom_q6v5.c
+++ b/drivers/remoteproc/qcom_q6v5.c
@@ -127,7 +127,7 @@ irqreturn_t q6v5_fatal_interrupt(int irq, void *data)
 	if (!q6v5->running)
 		return IRQ_HANDLED;
 
-	msg = qcom_smem_get(QCOM_SMEM_HOST_ANY, q6v5->crash_reason, &len);
+	msg = qcom_smem_get(q6v5->remote_id, q6v5->crash_reason, &len);
 	if (!IS_ERR(msg) && len > 0 && msg[0])
 		dev_err(q6v5->dev, "fatal error received: %s\n", msg);
 	else
-- 
2.34.1

